cohortextractor/expectation_generators.py

```diff
@@ -138,10 +140,21 @@ def generate(population, **kwargs):
 
     if match_incidence is not None:
         # Remove rows to match the incidence of the passed-in series
-        df.loc[pd.isnull(match_incidence), :] = None
+        set_empty_values(df, pd.isnull(match_incidence))
     elif not universal:
         # Randomly remove rows to match incidence
-        df.loc[df.sample(n=int((1 - incidence) * population)).index, :] = None
+        set_empty_values(df, df.sample(n=int((1 - incidence) * population)).index)
     if date is None:
         df = df.drop("date", axis=1)
     return df
+
+
+def set_empty_values(df, row_selection):
+    empty_values = {
+        "bool": False,
+        "int": 0,
+        "float": 0.0,
+        "date": None,
```

Comment:
Is this correct? This suggests date should be empty string:

https://github.com/opensafely/cohort-extractor/blob/c4d1fc741d59e5ae7e2d585e6db880608e01a44c/cohortextractor/emis_backend.py#L173-L185

Or is "default" different from "empty"?

Reply:
Ah yes! This is intentional and I meant to add a comment here but forgot. I'll update.